lib/matplotlib/lines.py

```diff
@@ -753,6 +753,12 @@ def draw(self, renderer):
             else:
                 gc.set_foreground(edgecolor)
                 gc.set_linewidth(self._markeredgewidth)
+                mec = self._markeredgecolor
+                if (is_string_like(mec) and mec == 'auto' and
+                    rgbaFace is not None):
+                    gc.set_alpha(rgbaFace[3])
+                else:
+                    gc.set_alpha(self.get_alpha())
```

Comment:
Not too sure about this line.  If we have set an alpha, then this will replace the alpha on the marker edge colour.  This feels wrong as a more specific value should take precedence over a more general value, right?

As I see it we should multiply alpha values here, so we do something like:

```python
edgecolor[3] *= self.get_alpha()
```
and the same for the other colours...